pkg/depend/write.go

```diff
@@ -27,5 +46,50 @@ func (lock *Lock) WriteLock(node *Node) error {
 		return err
 	}
 
-	return lock.cache.PushNotes(node.url, ref_lock_name)
+	if err = lock.cache.PushNotes(node.url, ref_lock_name); err != nil {
+		return err
+	}
+
+	if err = lock.populateLockRef(node); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (lock *lock) removeLock(node *Node) error {
+	if err := lock.populateLockRef(node); err != nil {
+		return err
+	}
+	if err := lock.cache.RemoveNotes(node.url, ref_lock_name, lock.lockref.object); err != nil {
+		return err
+	}
+	if err := lock.cache.PushNotes(node.url, ref_lock_name); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (lock *lock) populateLockRef(node *Node) error {
+	byte_s, err := lock.cache.ListNotes(node.url, ref_lock_name)
+	if err != nil {
+		return err
+	}
+	// For Windows compatability
+	noteRefs := strings.Split(strings.TrimSpace(strings.ReplaceAll(string(byte_s), "\r\n", "\n")), "\n")
+	if len(noteRefs) > 1 {
+		lastRef := strings.Fields(noteRefs[len(noteRefs)-1])[1]
+		// TODO: This should be managed by an appropriate logging library
+		fmt.Sprintln("WARN: Number of locks in the repo > 1, lock used maps to git SHA", lastRef)
```

Comment:
Should this be a warning or error?

Reply:
Yeh you're right., I think it's better to get this to fail loudly and just error out